### ISSUE TITLE
How to change output type of Core ML Model (Keras)

### ISSUE THREAD

**Issue by anooprawat1** [hwchong/MNIST_DRAW#1]
Hi Nice tutorial ! I also implemented same few days back and manage to add input to model as CVPixelBuffer. I have one question.. How can I make the output of ml model as number which it predicts and with probability instead of type MultiArray<Double,10>(Like flower example in WWDC). Should we have to create a custom keras layer class or use lambda layer?. I had search enough but couldnt find anything.

**Comment by hwchong** [comment]
Hello and thank you for liking the tutorial! You shouldn't have to create another Keras layer or to use a lambda layer. The model was trained with an output vector of 10 elements, each corresponding to the Y-Label of the MNIST dataset.

MultiArray<Double,10> is exactly the same as the output vector in the Python Keras model. If you debug the app and inspect the output you will see that it will have only 10 elements in it and each with a double value. The double value is the probability obtained after having run a softmax classifier on the results. 

Please refer to the model's source in this notebook - https://github.com/hwchong/MNIST_DRAW/blob/master/Training/notebooks/Keras-1.2.2-mnist-cnn.ipynb

I hope that answers your question.

**Comment by anooprawat1** [comment]
@hwchong I got this but my question.How to create the output as Int , instead of vector of 10 element(MultiArray<Double,10>). Like in your app if I draw 2 it should gives me output 2 (as Int) not [0,0,1,0,0,0,0,0,0,0](MultiArray<Double,10>).
Like in Flower example screenshot . The output is not MultiArray , it gives probabilities and name of flower.
<img width="829" alt="screen shot 2017-07-11 at 10 31 36 am" src="https://user-images.githubusercontent.com/26578409/28051389-a29b9020-6621-11e7-9197-381fbd7be81d.png">

**Comment by hwchong** [comment]
Well....it's only one line of code to convert an array to a Dictionary. If you really want the CoreML tool to do it for you then you have to specify that to the coremltool converter. It's documented here: 

http://pythonhosted.org/coremltools/generated/coremltools.converters.keras.convert.html#coremltools.converters.keras.convert
